docs/server/code.md

```diff
@@ -0,0 +1,135 @@
+## Code
+
+This document will cover the design principals that should guide the server side component for the MCP.
+
+The code development is guided as much as possible by the principals behind Onion architecture or Clean architecture:
```

Comment:
Are there any videos/resources we could link out to for a more in-depth overview of the architecture?

Reply:
+1 on this. Linking to existing stuff would be useful